components/proxy/src/main/kotlin/com/hotels/styx/ObjectTags.kt

```diff
@@ -15,15 +15,58 @@
  */
 package com.hotels.styx
 
+private const val LBGROUP = "lbGroup"
+private val LBGROUP_REGEX = "$LBGROUP=(.+)".toRegex()
 fun lbGroupTag(name: String) = "lbGroup=$name"
-
-fun lbGroupTagValue(tag: String): String? = "lbGroup=(.+)".toRegex()
-        .matchEntire(tag)
+fun lbGroupTag(tags: Set<String>) = tags.firstOrNull(::isLbGroupTag)
+fun isLbGroupTag(tag: String) = LBGROUP_REGEX.matches(tag)
+fun lbGroupTagValue(tags: Set<String>) = lbGroupTagValue(lbGroupTag(tags)?:"")
+fun lbGroupTagValue(tag: String): String? = LBGROUP_REGEX.matchEntire(tag)
         ?.groupValues
         ?.get(1)
 
 fun sourceTag(creator: String) = "source=$creator"
-
 fun sourceTag(tags: Set<String>) = tags.firstOrNull { it.startsWith("source=") }
-
 fun sourceTagValue(tags: Set<String>) = sourceTag(tags)?.substring("source".length + 1)
+
+private const val STATE = "state"
+const val STATE_ACTIVE = "active"
+const val STATE_UNREACHABLE = "unreachable"
+const val STATE_CLOSED = "closed"
```

Comment:
This is really `inactive` that is different from `closed`:
- inactive: the resources are preserved but object is out of load balancer
- closed: the resources are removed and object is due to be removed for good